src/test/java/com/jcabi/manifests/ManifestsTest.java

```diff
@@ -127,4 +130,95 @@ public void appendsAttributesFromInputStream() throws Exception {
         );
     }
 
+    /**
+     * Manifests can get the first value when multiple manifests with the same
+     * attribute have been appended.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void getGetsTheFirstEntryAcrossMultipleManifests() throws Exception {
+        final String name = "Attr";
+        final Manifests mfs = new Manifests();
+        mfs.append(ManifestsTest.manifestStream("Attr: a\n"));
+        mfs.append(ManifestsTest.manifestStream("Attr: b\n"));
+        MatcherAssert.assertThat(
+            "The value is from the first manifest",
+            mfs.containsKey(name) && "a".equals(mfs.get(name))
+        );
+    }
+
+    /**
+     * Manifests can get allValues of an attribute across multiple appended
+     * manifests.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void allValuesGetsAListOfAllValuesOfSameKey() throws Exception {
+        final Manifests mfs = new Manifests();
+        mfs.append(ManifestsTest.manifestStream("Multi-Value-Attr: 1\n"));
+        mfs.append(ManifestsTest.manifestStream("Multi-Value-Attr: 2\n"));
+        MatcherAssert.assertThat(
+            mfs.allValues("Multi-Value-Attr"),
+            Matchers.contains("1", "2")
+        );
+    }
+
+    /**
+     * Manifest can get allValues of two attributes across multiple appended
+     * manifests.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void allValuesGetsAListOfAllValues() throws Exception {
+        final Manifests mfs = new Manifests();
+        mfs.append(ManifestsTest.manifestStream("Attr1: 1a\n"));
+        mfs.append(ManifestsTest.manifestStream("Attr1: 1b\nAttr2: 2a\n"));
+        mfs.append(ManifestsTest.manifestStream("Attr1: 1c\n"));
+        mfs.append(ManifestsTest.manifestStream("Attr2: 2b\n"));
+        MatcherAssert.assertThat(
+            mfs.allValues("Attr1"),
+            Matchers.contains("1a", "1b", "1c")
+        );
+        MatcherAssert.assertThat(
+            mfs.allValues("Attr2"),
+            Matchers.contains("2a", "2b")
+        );
+    }
+
+    /**
+     * Manifest can return a list of all values that will not get mutated by
+     * further append operations.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void listReturnedByAllValuesIsACopy() throws Exception {
```

Comment:
@pinaf Well it's not immutable, it's just your own copy, an independent reference to anything held inside Manifest. I will go for `returnsIndependentListOfAllValues`